### ISSUE TITLE
Use R6 objects instead of S4

### ISSUE THREAD

**Issue by PavelBal** [Open-Systems-Pharmacology/rClr#31]

**Comment by PavelBal** [comment]
tbd

**Comment by PavelBal** [comment]
I would like to discuss the idea of using R6 classes in the package. Currently, what is returned by rClr when loading .NET object is a S4 object:

```
> sim <- loadTestSimulation("S1")
> sim$ref
An object of class "cobjRef"
Slot "clrobj":
<pointer: 0x000001ed7716c3a0>

Slot "clrtype":
[1] "OSPSuite.R.Domain.Simulation"
```

The S4 class is defined here https://github.com/Open-Systems-Pharmacology/rClr/blob/rClr-R/R/00classDefinitions.R

We have gathered much experience with R6 OOP over the years, and I think using R6 we could provide a more intuitive interface to .NET, as R6 is closer to OOP in C++/C#. We could apply some of our convenience functions from ospsuite (== move them to `rNET`). These are the pros.

The cons are:
- As we found out, most users are not so familiar with R6. However in this case, I do not see it as a problem, as users of this package are most probably very familiar with C# and its OOP.
- This would be completely incompatible with `rClr`, and we would have to re-write many things in `ospsuite`. However, I see it as a (challenging) opportunity rather than a problem.

@msevestre @Yuri05 Would like to hear your thoughts.
